catalog/task/collect-extra-data/0.2/collect-extra-data.yaml

```diff
@@ -0,0 +1,101 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: collect-extra-data
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to collect extra data from release resources and optionally save CRs to a workspace
+  params:
+    - name: release
+      type: string
+      description: The namespaced name of the Release
+    - name: releasePlan
+      type: string
+      description: The namespaced name of the ReleasePlan
+    - name: releasePlanAdmission
+      type: string
+      description: The namespaced name of the ReleasePlanAdmission
+    - name: snapshot
+      type: string
+      description: The namespaced name of the Snapshot
+    - name: saveRelease
+      type: string
+      description: Save the Release CR as a json in the data workspace
+      default: true
+    - name: saveReleasePlan
+      type: string
+      description: Save the ReleasePlan CR as a json in the data workspace
+      default: true
+    - name: saveReleasePlanAdmission
+      type: string
+      description: Save the ReleasePlanAdmission CR as a json in the data workspace
+      default: true
+    - name: saveSnapshot
+      type: string
+      description: Save the Snapshot CR spec as a json in the data workspace
+      default: true
+  workspaces:
+    - name: data
+      description: Workspace to save the CR jsons to
+  results:
+    - name: extraData
+      description: |
+        Result of merging the `extraData` field of all the release resources
+  steps:
+    - name: collect-extra-data
+      image: quay.io/hacbs-release/release-utils:f486f350df43974706fc484a13bfeefc2c8fa95a
+      env:
+        - name: "RELEASE"
+          value: '$(params.release)'
+        - name: "RELEASE_PLAN"
+          value: '$(params.releasePlan)'
+        - name: "RELEASE_PLAN_ADMISSION"
+          value: '$(params.releasePlanAdmission)'
+        - name: "SNAPSHOT"
+          value: '$(params.snapshot)'
+      script: |
+        #!/usr/bin/env sh
```

Comment:
I think you will also need at least `set -e`. See here: https://tekton.dev/docs/pipelines/tasks/#running-scripts-within-steps

>Scripts that do not start with a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)) line will have the following default preamble prepended:

```
#!/bin/sh
set -e
```

So I think this implies that if you have a shebang, then there's nothing. And I think bash will not have this enabled by default. But I might be wrong.

Reply:
This could not work in some cases as it exists if there's an error and sometimes we want to handle the error ourselves like in this script.

Reply:
> sometimes we want to handle the error ourselves like in this script.

Oh, sorry, I read it wrong. The `exit 1` ensures that the script exits regardless of `set +e` or `set -e`, you're right. So if that's the intention here - that we should ignore any issues along the way - then disregard this. Personally, I'd try to avoid this - for example if in the middle of the script execution something goes wrong with the cluster and you cannot fetch one of the resources (that don't have the explicit error handling), I would want the pipeline to fail...

Reply:
Resolving this as the code in question was just copied from the first version of the task. If/when we change things related to the extraData field name and how it is structured, we can revisit it